rust/src/lib/query_apply/net_state.rs

```diff
@@ -119,9 +119,30 @@ impl NetworkState {
             }
         }
         if !self.kernel_only {
-            let nm_state = nm_retrieve(self.running_config_only).await?;
+            let mut is_forwarding_supported = false;
+            let nm_state = nm_retrieve(
+                self.running_config_only,
+                &mut is_forwarding_supported,
+            )
+            .await?;
             // TODO: Priority handling
             self.update_state(&nm_state);
+
+            // Clear forwarding when unsupported by NetworkManager to prevent failures
+            // if the user tries to apply the returned state as-is.
+            // TODO: Remove this code once the minimum supported NetworkManager version is >= 1.54
+            if !is_forwarding_supported {
+                self.interfaces
+                    .user_ifaces
+                    .values_mut()
+                    .chain(self.interfaces.kernel_ifaces.values_mut())
+                    .for_each(|iface| {
+                        if let Some(ipv4) = iface.base_iface_mut().ipv4.as_mut()
+                        {
+                            ipv4.forwarding = None;
+                        }
+                    });
```

Comment:
Please do not put merging code at this level, otherwise code are spreading everything in the project. Nmstate has 100+ properties, why none of them need merge code at `rust/src/lib/query_apply/net_state.rs`?

Can you spend on sometime understand how nmstate merging nispor query output with nm_retrieve query output?

With that you will understand how this forwarding should be merged.

Meanwhile, there are a lot merged patch adding properties to nmstate schema, they could be very good example on how to do it.